ui/src/wallet.ts

```diff
@@ -91,6 +72,40 @@ export function build(
     status: Status.NotConnected,
   });
 
+  function onModalHide(): void {
+    const store = svelteStore.get(stateStore);
+    if (store.status === Status.NotConnected) {
+      reinitWalletConnect();
+    }
+  }
+
+  function newWalletConnect(): WalletConnect {
+    return new WalletConnect({
+      bridge: "https://bridge.walletconnect.org",
+      qrcodeModal: qrCodeModal,
+    });
+  }
+
+  // We need to reinitialize `WalletConnect` until this issue is fixed:
+  // https://github.com/WalletConnect/walletconnect-monorepo/pull/370
```

Comment:
Looks like the referenced PR got merged, maybe we can upgrade walletconnect and this won't be necessary?

Reply:
The PR is merged but the issue isn't fixed :) It's all in the PR, have a look for your interest.

Reply:
Too bad.